sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/cases/CaseDetailedTableViewSteps.java

```diff
@@ -0,0 +1,217 @@
+          AtomicInteger atomicInt = new AtomicInteger();
+          List<WebElement> tableData = table.findElements(CASE_DETAILED_TABLE_DATA);
+          tableData.forEach(
+              dataText -> {
+                webDriverHelpers.scrollToElementUntilIsVisible(dataText);
+                indexWithData.put(atomicInt.getAndIncrement(), dataText.getText());
+              });
+          tableDataList.add(indexWithData);
+        });
+    List<Map<String, String>> tableObjects = new ArrayList<>();
+    tableDataList.forEach(
+        row -> {
+          ConcurrentHashMap<String, String> objects = new ConcurrentHashMap<>();
+          headers.forEach((headerText, index) -> objects.put(headerText, row.get(index)));
+          tableObjects.add(objects);
+        });
+    return tableObjects;
+  }
+
+  private List<WebElement> getTableRows() {
+    webDriverHelpers.waitUntilIdentifiedElementIsVisibleAndClickable(CASE_DETAILED_COLUMN_HEADERS);
+    return baseSteps.getDriver().findElements(CASE_DETAILED_FIRST_TABLE_ROW);
+  }
+
+  private Map<String, Integer> extractColumnHeadersHashMap() {
+    AtomicInteger atomicInt = new AtomicInteger();
+    HashMap<String, Integer> headerHashmap = new HashMap<>();
+    webDriverHelpers.waitUntilIdentifiedElementIsVisibleAndClickable(CASE_DETAILED_COLUMN_HEADERS);
+    webDriverHelpers.waitUntilAListOfWebElementsAreNotEmpty(CASE_DETAILED_COLUMN_HEADERS);
+    baseSteps
+        .getDriver()
+        .findElements(CASE_DETAILED_COLUMN_HEADERS)
+        .forEach(
+            webElement -> {
+              webDriverHelpers.scrollToElementUntilIsVisible(webElement);
+              headerHashmap.put(webElement.getText(), atomicInt.getAndIncrement());
+            });
+    return headerHashmap;
+  }
+
+  /* Used in test: Check Case details in Detailed table view from Case directory
+  Returns Case report date as displayed in the UI
+  <parameter> dateTimeString: represents the date time value read from Case created through API <parameter>
+  */
+
+  public String getDetaOfReportDateTime(String dateTimeString) {
+    SimpleDateFormat outputFormat = new SimpleDateFormat("M/dd/yyyy h:mm a");
+    // because API request is sending local GMT and UI displays GMT+2 (server GMT)
+    outputFormat.setTimeZone(TimeZone.getTimeZone("GMT+2"));
+
+    // inputFormat is the format of teh dateTime as read from API created case
+    DateFormat inputFormat = new SimpleDateFormat("EEE MMM dd HH:mm:ss zzz yyyy", Locale.US);
+    Date parsedDate = null;
+    try {
+      parsedDate = inputFormat.parse(dateTimeString);
+    } catch (ParseException e) {
+      e.printStackTrace();
+    }
+    return outputFormat.format(parsedDate);
+  }
+
+  public String getFollowUpUntilCaseDate(String dateOfReportDateDateTime) {
+    SimpleDateFormat outputFormat = new SimpleDateFormat("M/dd/yyyy");
+    DateFormat inputFormat = new SimpleDateFormat("MM/dd/yyyy h:mm a");
+    ZoneId defaultZoneId = ZoneId.systemDefault();
```

Comment:
move variable in class

Reply:
This variable is already in class: apiState.getCreatedCase().getReportDate()